llvm/lib/Transforms/Scalar/ConstraintElimination.cpp

```diff
@@ -1048,6 +1053,28 @@ void State::addInfoForInductions(BasicBlock &BB) {
   }
 }
 
+void State::addInfoForInductions(BasicBlock &BB) {
+  auto *L = LI.getLoopFor(&BB);
+  if (!L)
+    return;
+  if (L->getHeader() != &BB)
+    return;
+
+  BasicBlock *Curr = &BB;
+  while (L->isLoopExiting(Curr)) {
```

Comment:
This looks pretty weird to me -- how is the fact that it's an exit chain from the header relevant?  I'd expect us to either inspect all exits or all latch-dominating exits (I'd have to look closer to check which one is correct).